rasa/core/evaluation/markers.py

```diff
@@ -56,21 +56,18 @@ def from_file(cls, path: Text) -> Dict:
     def from_yaml(cls, yaml: Text, filename: Text = "") -> Dict:
         """Loads the config from YAML text after validating it."""
         try:
-            # TODO implement and call the yaml schema validation
-            #  rasa.shared.utils.validation.validate_yaml_schema(yaml,
-            #  MAKERS_SCHEMA_FILE)
+            config = rasa.shared.utils.io.read_yaml(yaml)
+            cls.validate_config(config, filename)
+            return config
 
-            return rasa.shared.utils.io.read_yaml(yaml)
         except ParserError as e:
             e.filename = filename
-            rasa.shared.utils.io.raise_warning(
-                message=f"The file {filename} could not be loaded "
+            raise YamlSyntaxException(
+                f"The file {filename} could not be loaded "
```

Comment:
I actually think the constructor of `YamlSyntaxEception` only accepts `filename` and `e` arguments, and not message, since this exception class has a custom `__str__` method. Otheriwse this string you've added as message will be considered the `filename` in its constructor and the message could end up being confusing.